scripts/ci/pr-bot/processPrUpdate.ts

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const github = require("@actions/github");
+const commentStrings = require("./shared/commentStrings");
+const { processCommand } = require("./shared/userCommand");
+const { addPrComment, nextActionReviewers } = require("./shared/githubUtils");
+const { PersistentState } = require("./shared/persistentState");
+const { ReviewerConfig } = require("./shared/reviewerConfig");
+const { PATH_TO_CONFIG_FILE } = require("./shared/constants");
+
+async function areReviewersAssigned(
+  pullNumber: number,
+  stateClient: typeof PersistentState
+): Promise<boolean> {
+  const prState = await stateClient.getPrState(pullNumber);
+  return Object.values(prState.reviewersAssignedForLabels).length > 0;
+}
+
+async function processPrComment(
+  payload,
+  stateClient: typeof PersistentState,
+  reviewerConfig: typeof ReviewerConfig
+) {
+  const commentContents = payload.comment.body;
+  const commentAuthor = payload.sender.login;
+  console.log(commentContents);
+  if (
+    await processCommand(
+      payload,
+      commentAuthor,
+      commentContents,
+      stateClient,
+      reviewerConfig
+    )
+  ) {
+    // If we've processed a command, don't worry about trying to change the attention set.
+    // This is not a meaningful push or comment from the author.
+    console.log("Processed command");
+    return;
+  }
+
+  // If comment was from the author, we should shift attention back to the reviewers.
+  console.log(
+    "No command to be processed, checking if we should shift attention to reviewers"
+  );
+  const pullAuthor =
```

Comment:
I've seen this author access pattern in a different review. If this pattern is repeated, consider adding a helper method.